The program does not call the "give_shell" function which could print the contents of flag.txt. I found the address of the "give_shell" using gdb and it is 0x08048581. Also, there is buffer overflow vulnerability in "vuln" function. I exploited this vulnerability to overwrite its return address with the address of "give_shell". "give_shell" calls system with its argument as the first command argument of program. Therefore, the final command is "./OverRun 'more flag.txt' $(python -c 'print ("A"*28 + "\x81\x85\x04\x08")')". The flag is "cs628a{edc1d08aad1ba15a1ca9f63dcdf62d2d}".
